fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -766,6 +781,10 @@ private boolean checkTxnHasRelatedJob(TransactionState txnState, Map<Long, Set<L
             return true;
         }
 
+        if (txnState.getSourceType() == LoadJobSourceType.ROUTINE_LOAD_TASK) {
+            return txnState.getCallBack().checkTxnHasRelatedJob(txnState);
```

Comment:
Do you need to callback for check relate job?

Reply:
Yes, it need. The reason is that task will be add into a needSchedulerQueue when it is created. Before task scheduler to be, the txn  state of task  is PREPARE. At the same time,  GlobelTransactionMgr clean old transactions include PREPARE. The task and txn of task live and die together